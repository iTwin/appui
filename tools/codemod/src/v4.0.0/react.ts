/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import type { API, FileInfo, Options } from "jscodeshift";
import { useExtensions } from "../utils/Extensions";

export default function transformer(
  file: FileInfo,
  api: API,
  options: Options
) {
  const j = api.jscodeshift;
  useExtensions(j);

  const root = j(file.source);

  root.rename(
    "@itwin/appui-react:CommonWidgetProps",
    "@itwin/appui-react:Widget"
  );
  root.rename("@itwin/appui-react:WidgetProps", "@itwin/appui-react:Widget");

  return root.toSource(options.printOptions);
}
