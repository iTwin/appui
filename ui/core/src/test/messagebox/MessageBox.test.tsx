/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";
import { expect } from "chai";

import { MessageBox, MessageSeverity, ButtonType, ButtonStyle } from "../../index";
import TestUtils from "../TestUtils";

describe("MessageBox", () => {

  before(async () => {
    await TestUtils.initializeUiCore();
  });

  const buttonCluster = [
    { type: ButtonType.Close, buttonStyle: ButtonStyle.Primary, onClick: () => { } },
  ];

  describe("renders", () => {
    it("should render", () => {
      const wrapper = mount(<MessageBox opened={true} severity={MessageSeverity.Information} buttonCluster={buttonCluster} />);
      const icon = wrapper.find("div.message-box-information");
      expect(icon.length).to.eq(1);
      wrapper.unmount();
    });

    it("renders correctly", () => {
      shallow(<MessageBox opened={true} severity={MessageSeverity.Information} buttonCluster={buttonCluster} />).should.matchSnapshot();
    });
  });

  describe("renders different severities", () => {
    it("MessageSeverity.Question", () => {
      const wrapper = mount(<MessageBox opened={true} severity={MessageSeverity.Question} buttonCluster={buttonCluster} />);
      const icon = wrapper.find("div.message-box-question");
      expect(icon.length).to.eq(1);
    });
    it("MessageSeverity.Warning", () => {
      const wrapper = mount(<MessageBox opened={true} severity={MessageSeverity.Warning} buttonCluster={buttonCluster} />);
      const icon = wrapper.find("div.message-box-warning");
      expect(icon.length).to.eq(1);
    });
    it("MessageSeverity.Error", () => {
      const wrapper = mount(<MessageBox opened={true} severity={MessageSeverity.Error} buttonCluster={buttonCluster} />);
      const icon = wrapper.find("div.message-box-error");
      expect(icon.length).to.eq(1);
    });
    it("MessageSeverity.Fatal", () => {
      const wrapper = mount(<MessageBox opened={true} severity={MessageSeverity.Fatal} buttonCluster={buttonCluster} />);
      const icon = wrapper.find("div.message-box-fatal");
      expect(icon.length).to.eq(1);
    });
  });

});
