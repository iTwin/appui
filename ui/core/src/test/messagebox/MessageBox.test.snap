// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders renders correctly 1`] = `
<Dialog
  alignment="center"
  buttonCluster={
    Array [
      Object {
        "buttonStyle": "uicore-buttons-primary",
        "onClick": [Function],
        "type": "close",
      },
    ]
  }
  inset={true}
  minHeight={400}
  minWidth={400}
  modal={true}
  movable={false}
  opened={true}
  resizable={false}
  width="512px"
>
  <MessageContainer
    severity={1}
  />
</Dialog>
`;
